src/main/java/org/mvplugins/multiverse/core/destination/Destination.java

```diff
@@ -8,7 +8,7 @@
 import org.jvnet.hk2.annotations.Contract;
 
 @Contract
-public interface Destination<T extends DestinationInstance> {
+public interface Destination<D extends Destination<D, T>, T extends DestinationInstance<T, D>> {
```

Comment:
Not crazy about the cursive generics here. Can you summarize what this is doing? Having some difficulty following.

Reply:
Its just each Destination maps a one DestinationInstance

Reply:
This is so that `ExactDestination.parseDestination()` can only return `ExactDestinationInstance`, then `ExactDestinationInstance.getDestination()` can only return `ExactDestination`. 

Kinda  like a 1 to 1 map thing, where you cant just allow ExactDestination to return any type of DestinationInstance and ExactDestinationInstance be created by any type of Destination